driver-kotlin-sync/src/main/kotlin/com/mongodb/kotlin/client/MongoCursor.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kotlin.client
+
+import com.mongodb.ServerAddress
+import com.mongodb.ServerCursor
+import com.mongodb.client.MongoCursor as JMongoCursor
+import java.io.Closeable
+
+/**
+ * The Mongo Cursor interface implementing the iterator protocol.
+ *
+ * An application should ensure that a cursor is closed in all circumstances, e.g. using a `use` statement:
+ * ```
+ *  collection.find().cursor().use { c ->
+ *      while (c.hasNext()) {
+ *          println(c.next())
+ *      }
+ *  }
+ * ```
+ *
+ * @param T The type of documents the cursor contains
+ * @property wrapped the underlying sync cursor
+ */
+public open class MongoCursor<T : Any>(private val wrapped: JMongoCursor<T>) : Iterator<T>, Closeable {
```

Comment:
Why is this class open for extension? Is it because of `MongoChangeStreamCursor`?  If so, should we make this class `sealed` instead?

Reply:
sealed classes cannot be initiated but will add a base cursor type.

Reply:
Done!